app/src/main/java/com/nowak/wjw/simplecompass/ui/main/MainViewModel.java

```diff
@@ -35,19 +40,93 @@
     public LiveData<Integer> buttonTextResId;
     public LiveData<Boolean> editTextVisible;
     public LiveData<Boolean> hideKeyBoard;
+    public LiveData<Boolean> showLocationError;
+    private MutableLiveData<Boolean> mShouldProvideRationale = new MutableLiveData<>(false);
+    public LiveData<Event<Boolean>> shouldProvideRationale = Transformations.map(mShouldProvideRationale, should -> new Event(should));
+    private MutableLiveData<Boolean> mShowPermissionDenied = new MutableLiveData<>(false);
+    public LiveData<Event<Boolean>> showPermissionDenied = Transformations.map(mShowPermissionDenied, show -> new Event<>(show));
+    private MutableLiveData<Boolean> mRequestLocationPermission = new MutableLiveData<>(false);
+    public LiveData<Event<Boolean>> requestLocationPermission = Transformations.map(mRequestLocationPermission, request -> new Event<>(request));
+
+
+    public MainViewModel(GetAzimuthUseCase getAzimuthUseCase, StartStopSensorListenerUseCase startStopSensorListenerUseCase, InitiateLastLocationUseCase initiateLastLocationUseCase, RequestAndStopLocationUpdatesUseCase requestAndStopLocationUpdatesUseCase, GetDestinationBearingUseCase getDestinationBearingUseCase) {
```

Comment:
ten konstruktor jest tragicznie długi, samo formatowanie jużby pomogło.

Reply:
co do tego konstruktora to mam pytanie - czy inicjalizować te zmienne w konstruktorze, czy lepiej utworzyć metody które będą je ekspozycjonować,  np 

 public LiveData<Float> getDestArrowRotation(){
      return  Transformations.switchMap(mDestinationBearing, b -> {
            Timber.d("destArrowRotation");
            if (b == null) return null;
            return Transformations.map(needleRotation, r -> {
                Timber.d("needleRotation");
                return r + b;
            });
        });

tylko, że wtedy za każdym razem kiedy wywołany tę metodę to będziemy na nowo wywoływać ten kod, a tak to tylko raz go wywołujemy - plus Transformacje go wywołują, ale to i tak by się działo...

Reply:
po to masz blok inicjalizujący init żeby takie startowe operacje wykonać w momencie użycia viewModelu.